search-commons/src/test/java/no/unit/nva/search/OpensearchTest.java

```diff
@@ -47,41 +49,76 @@ public class ElasticsearchTest {
     public static final String STATUS_TO_INCLUDE_IN_RESULT = "Pending";
     public static final int ZERO_HITS_BECAUSE_APPROVED_WAS_FILTERED_OUT = 0;
     public static final long DELAY_AFTER_INDEXING = 1000L;
-    private static final String ELASTICSEARCH_VERSION = "7.10.2";
     private static final int PAGE_SIZE = 10;
     private static final int PAGE_NO = 0;
     private static final URI ORGANIZATION_ID_URI_HARDCODED_IN_SAMPLE_FILES = URI.create("https://www.example.com/20754.0.0.0");
     private static final String COMPLETED = "Completed";
-    
-    @Container
-    public ElasticsearchContainer container = new ElasticsearchContainer(DockerImageName
-                                                                             .parse(ELASTICSEARCH_OSS)
-                                                                             .withTag(ELASTICSEARCH_VERSION));
+    private static final int CONTAINER_PORT = 9200;
+
     private SearchClient searchClient;
     private IndexingClient indexingClient;
 
+
+    @SuppressWarnings("rawtypes")
+    private final GenericContainer container = new GenericContainer(
+            DockerImageName.parse("opensearchproject/opensearch:latest"))
+            .withEnv("plugins.security.disabled", "true")
+            .withEnv("discovery.type", "single-node");
+
+    @SuppressWarnings("unchecked")
     @BeforeEach
     void setUp() {
-        RestClientBuilder restClientBuilder = RestClient.builder(HttpHost.create(container.getHttpHostAddress()));
+
+        container.setPortBindings(List.of(CONTAINER_PORT + ":" + CONTAINER_PORT));
+        container.setExposedPorts(List.of(CONTAINER_PORT));
+        container.start();
+
+        var httpHostAddress = getContainerHost();
+
+        RestClientBuilder restClientBuilder = RestClient.builder(HttpHost.create(httpHostAddress));
         RestHighLevelClientWrapper restHighLevelClientWrapper = new RestHighLevelClientWrapper(restClientBuilder);
 
         searchClient = new SearchClient(restHighLevelClientWrapper);
         indexingClient = new IndexingClient(restHighLevelClientWrapper);
     }
 
+    @AfterEach
+    void afterEach() {
+        container.stop();
+    }
+
+
+    @Test
+    void canConnectToContainer() throws IOException, InterruptedException{
+
+        var httpHostAddress = getContainerHost();
+
+        HttpClient httpClient = HttpClient.newBuilder().build();
+
+        HttpRequest request = HttpRequest.newBuilder()
+                .GET()
+                .uri(URI.create("http://" + httpHostAddress))
```

Comment:
At this point http:// should just be included in getContainerHost, or getContainerHost should return a URI

Reply:
getContainerHost() is used in setup as well. In the setup we dont expect an URI.